importer.go

```diff
@@ -14,10 +14,46 @@ import (
 	"sync"
 )
 
-var (
-	goPath = os.Getenv("GOPATH")
-	goSrc  = filepath.Join(goPath, "src")
-)
+// ErrNotInGoPath is an error returned when a package is not in any of the
+// possible go paths.
+var ErrNotInGoPath = fmt.Errorf("parseutil: package is not in any of the go paths")
+
+// GoPath is the collection of all go paths.
+type GoPath []string
+
+// Abs returns the absolute path to a package. The go path in the absolute path
+// that will be returned is the first that contains the given package.
+func (gp GoPath) Abs(pkg string) (string, error) {
+	for _, p := range gp {
+		path := filepath.Join(p, "src", pkg)
+		if _, err := os.Stat(path); os.IsNotExist(err) {
+			continue
+		} else if err != nil {
+			return "", err
+		} else {
+			return path, nil
```

Comment:
I'd found more readable if the most usual case would be the first, so it'd be:
```
func (gp GoPath) Abs(pkg string) (string, error) {
	for _, p := range gp {
		path := filepath.Join(p, "src", pkg)
 		if _, err := os.Stat(path); err == nil {
 			return path, nil
 		} else if !os.IsNotExist(err) {
 			return "", err
 		}
 	}
 	return "", ErrNotInGoPath
 }
```
That way, the `continue` would be not neccessary